### PR TITLE
Approve Sitter Application Procedure

### PR DESCRIPTION
**Pull request by RotendaBBD**

### CODE REVIEW

File: flyway/migrations/V16__ApproveSitterApplicationProc.sql [hunk 1]
```diff
@@ -0,0 +1,19 @@
+CREATE PROCEDURE ApproveSitterApplication
```
Comment (jennadunfordbbd) [inline flyway/migrations/V16__ApproveSitterApplicationProc.sql:1]: Needs delimiter statement at top of file.

`DELIMITER //`

File: flyway/migrations/V16__ApproveSitterApplicationProc.sql [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+    @applicationID uniqueidentifier
```
Comment (jennadunfordbbd) [inline flyway/migrations/V16__ApproveSitterApplicationProc.sql:2]: Should be:

`CREATE PROCEDURE ApproveSitterApplication(IN applicationID int)`

File: flyway/migrations/V16__ApproveSitterApplicationProc.sql [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+
+AS
```
Comment (jennadunfordbbd) [inline flyway/migrations/V16__ApproveSitterApplicationProc.sql:4]: Should be removed. (or migration will fail)

File: flyway/migrations/V16__ApproveSitterApplicationProc.sql [hunk 4]
```diff
@@ -0,0 +1,19 @@
...
+BEGIN  
+    UPDATE SitterApplications
+    SET applicationStatus = 'Approved'
+    WHERE PK_ApplicationID = @applicationID
+        AND applicationStatus = 'Pending';
```
Comment (jennadunfordbbd) [inline flyway/migrations/V16__ApproveSitterApplicationProc.sql:9]: This should be in a transaction

File: flyway/migrations/V16__ApproveSitterApplicationProc.sql [hunk 5]
```diff
@@ -0,0 +1,19 @@
...
+    
+    IF @@ROWCOUNT > 0
```
Comment (jennadunfordbbd) [inline flyway/migrations/V16__ApproveSitterApplicationProc.sql:11]: Should be `IF ROWCOUNT() > 0`

File: flyway/migrations/V16__ApproveSitterApplicationProc.sql [hunk 6]
```diff
@@ -0,0 +1,19 @@
+CREATE PROCEDURE ApproveSitterApplication
+    @applicationID uniqueidentifier
+
+AS
+BEGIN  
+    UPDATE SitterApplications
+    SET applicationStatus = 'Approved'
+    WHERE PK_ApplicationID = @applicationID
```
Comment (jennadunfordbbd) [inline flyway/migrations/V16__ApproveSitterApplicationProc.sql:8]: remove @ from applicationID

File: flyway/migrations/V16__ApproveSitterApplicationProc.sql [hunk 7]
```diff
@@ -0,0 +1,19 @@
...
+    
+    IF @@ROWCOUNT > 0
+    BEGIN  
+        PRINT 'Sitter Application approved';
+    END
+    ELSE
+    BEGIN  
+        PRINT 'Sitter application ID not found, already approved, or no changes made';
+    END
+END
```
Comment (jennadunfordbbd) [inline flyway/migrations/V16__ApproveSitterApplicationProc.sql:19]: Needs delimeter statement at end (replaces the last END you have).

`END//`

`DELIMITER ;`

File: flyway/migrations/V16__ApproveSitterApplicationProc.sql [hunk 8]
```diff
@@ -0,0 +1,19 @@
...
+        AND applicationStatus = 'Pending';
+    
+    IF @@ROWCOUNT > 0
+    BEGIN  
+        PRINT 'Sitter Application approved';
+    END
+    ELSE
+    BEGIN  
+        PRINT 'Sitter application ID not found, already approved, or no changes made';
+    END
```
Comment (jennadunfordbbd) [inline flyway/migrations/V16__ApproveSitterApplicationProc.sql:18]: Please restructure to follow MySQL 8 syntax for IF statements:
[MYSQL 8 Reference: IF statements](https://dev.mysql.com/doc/refman/8.0/en/if.html)

**Review by jennadunfordbbd — CHANGES_REQUESTED** [review]
Left some comments on the syntax.

**Comment by jennadunfordbbd** [comment]
Here is the script that seems to work fine with MYSQL 8:

```
DELIMITER //

CREATE PROCEDURE ApproveSitterApplication(IN applicationID INT)    

BEGIN 
    UPDATE SitterApplications
    SET applicationStatus = 'Approved'
    WHERE PK_ApplicationID = applicationID
        AND applicationStatus = 'Pending';
    
    IF ROW_COUNT() > 0 THEN
       SELECT 'Sitter Application approved' AS output_message;
    ELSE
        SELECT 'Sitter application ID not found, already approved, or no changes made' AS output_message;
    END IF;
    
END//

DELIMITER ;
```

If you would like you can just go ahead and update the code to this and update the PR

**Review by jennadunfordbbd — APPROVED** [review]
Thanks! Looks good :)

**Review by Lior-Becker — APPROVED** [review]

**Pull request merged**